src/main/java/Interactor.java

```diff
@@ -0,0 +1,167 @@
+import Duke.Task.Deadline;
+import Duke.Task.Event;
+import Duke.Task.Task;
+import Duke.Task.Todo;
+import Duke.exceptions.DukeEmptyListException;
+import Duke.exceptions.DukeException;
+import Duke.exceptions.NoSuchCommandException;
+
+import java.util.Scanner;
+
+public class Interactor {
+
+    Operations operations;
+    private String exitKeyword = "";
+    private String doneKeyword = "";
+    private String todoKeyword = "todo";
+    private String eventKeyword = "event";
+    private String deadlineKeyword = "deadline";
+    private String deleteKeyword = "delete";
+
+    public Interactor() {}
+
+    public void setOperations(Operations operations) {
+        this.operations = operations;
+    }
+
+    public void setExitKeyword(String exitKeyword){
+        this.exitKeyword = exitKeyword;
+    }
+
+    public void setDoneKeyword(String doneKeyword) {
+        this.doneKeyword = doneKeyword;
+    }
+
+
+    public void start() throws DukeException {
+        printGreeting();
+        printExitKeyword();
+
+        boolean toContinue = true;
+
+        Scanner scanner = new Scanner(System.in);
+
+        do {
+            try {
+                String currentInput = scanner.nextLine();
+
+
+                if (currentInput.equals(exitKeyword)) {
+                    printGoodbye();
+                    break;
+                } else if (currentInput.equals("list")) {
+                    operations.printDatabase();
+                }
+                // When user toggles done-ness of task
+                else if (beginsWith(currentInput, doneKeyword)) {
+                    int i = getNumberFromInput(currentInput);
+                    operations.setDone(i);
+                }
+
+                // handle user delete item
+                else if (beginsWith(currentInput, deleteKeyword)) {
+                    int i = getNumberFromInput(currentInput);
+                    operations.deleteFromDatabase(i);
+                    System.out.println("Deleted successfully " + i);
+                }
+
+                //*** check for Duke.Task.Todo, deadline, event creations ***
+                else if (beginsWith(currentInput, todoKeyword)) {
+                    Todo todo = handleTodoCreation(currentInput);
+                    operations.addToDatabase(todo);
+                    printTaskAddedMessage(todo);
+
+                } else if (beginsWith(currentInput, deadlineKeyword)) {
+                    Deadline deadline = handleDeadlineCreation(currentInput);
+                    operations.addToDatabase(deadline);
+                    printTaskAddedMessage(deadline);
+
```

Comment:
currently i am using if else for this part as well, have you think of using switch to replace this?